noir-projects/aztec-nr/aztec/src/messages/logs/utils.nr

```diff
@@ -1,16 +1,22 @@
-use crate::oracle::notes::{get_app_tag_as_sender, increment_app_tagging_secret_index_as_sender};
+use crate::oracle::notes::{
+    get_app_tag_as_sender, get_sender_for_tags, increment_app_tagging_secret_index_as_sender,
+};
 use dep::protocol_types::address::AztecAddress;
 
 pub(crate) fn prefix_with_tag<let L: u32>(
     log_without_tag: [Field; L],
-    sender: AztecAddress,
     recipient: AztecAddress,
 ) -> [Field; L + 1] {
     // Safety: we assume that the sender wants for the recipient to find the tagged note, and therefore that they will
     // cooperate and use the correct tag. Usage of a bad tag will result in the recipient not being able to find the
     // note automatically.
-    let tag = unsafe { get_app_tag_as_sender(sender, recipient) };
-    increment_app_tagging_secret_index_as_sender(sender, recipient);
+    let tag = unsafe {
+        let sender = get_sender_for_tags().expect(
+            f"Sender for tags is not set when emitting a private log. Set it by calling `set_sender_for_tags(...)`.",
+        );
+        increment_app_tagging_secret_index_as_sender(sender, recipient);
+        get_app_tag_as_sender(sender, recipient)
```

Comment:
We could save some constraints here by wrapping these 3 calls in 1 unconstrained context as noir spends some constraints on checking the type of unconstrained function's return value. I am just not sure if it's significant. Does the reviewer know?